app/src/main/java/ch/epfl/sdp/lobby/game/FirebaseGameLobbyRepository.kt

```diff
@@ -59,46 +62,47 @@ class FirebaseGameLobbyRepository : IGameLobbyRepository {
     }
 
     override fun getParticipations(gameId: String, cb: Callback<List<Participation>>) {
-        fs.collection(GAME_COLLECTION)
-                .document(gameId).collection(GAME_PARTICIPATION_COLLECTION).get()
-                .addOnSuccessListener { doc -> cb(doc.map { it.toObject<Participation>() }) }
+        fs.collection(GAME_COLLECTION).document(gameId).get()
+            .addOnSuccessListener { doc ->
+                cb(doc.toObject<Game>()!!.participation)
+            }
     }
 
     override fun getAdminId(gameId: String, cb: Callback<String>) {
         fs.collection(GAME_COLLECTION).document(gameId).get()
                 .addOnSuccessListener { cb(it.toObject<Game>()!!.adminID) }
     }
 
-    override fun changePlayerReady(gameId: String, uid: String) {
-        setPlayerReady(gameId, uid, true)
+    override fun changePlayerReady(gameId: String, uid: String, cb: UnitCallback) {
+        setPlayerReady(gameId, uid, true, cb)
     }
 
-    override fun setPlayerReady(gameId: String, uid: String, ready: Boolean) {
-        fs.collection(GAME_PARTICIPATION_COLLECTION).whereEqualTo("playerID", uid).get().addOnSuccessListener { documents ->
-            for (doc in documents) {
-                fs.collection(GAME_PARTICIPATION_COLLECTION).document(doc.id).update("ready", ready)
-            }
-        }
+    private fun setHelper(gameId: String, cb: UnitCallback, modifier: participationModifier){
```

Comment:
The name is not very clear. When I read setHelper, I don't get what this function does. Maybe something like "updateUserParticipation" ?